src/main/resources/com/google/api/codegen/java/mock_service_impl.snip

```diff
@@ -52,23 +56,35 @@
   @@Override
   public void {@method.name}({@method.requestTypeName} request,
     StreamObserver<{@method.responseTypeName}> responseObserver) {
-    {@method.responseTypeName} response = ({@method.responseTypeName}) responses.remove();
-    requests.add(request);
-    responseObserver.onNext(response);
-    responseObserver.onCompleted();
+    Object response = responses.remove();
+    if (response instanceof {@method.responseTypeName}) {
+      requests.add(request);
+      responseObserver.onNext(({@method.responseTypeName}) response);
+      responseObserver.onCompleted();
+    } else if(response instanceof Exception) {
+      responseObserver.onError((Exception) response);
+    } else {
+      throw new IllegalArgumentException("Unrecognized response type");
```

Comment:
I switched to use` responseObserver.onError()` so an exception would be raised in the corresponding test.